t3f/autodiff_test.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+import tensorflow as tf
+
+from t3f import ops
+from t3f import initializers
+from t3f import variables
+from t3f import riemannian
+from t3f import autodiff
+
+
+class _AutodiffTest():
+
+  def testGradients(self):
+    w = initializers.random_matrix(([5] * 3, None), dtype=self.dtype)
+    A = initializers.random_matrix(([5] * 3, [5] * 3), dtype=self.dtype)
+    x = initializers.random_matrix(([5] * 3, None), dtype=self.dtype)
+    z = initializers.random_matrix(([5] * 3, None), dtype=self.dtype)
+
+    def func1(x):
+      return 0.5 * ops.flat_inner(x, w) ** 2
+
+    actual1 = ops.full(autodiff.gradients(func1, x))
+    desired1 = ops.full(ops.flat_inner(x, w) * riemannian.project(w, x))
+    with self.test_session() as sess:
+      actual1_v, desired1_v = sess.run([actual1, desired1])
+      np.testing.assert_allclose(actual1_v, desired1_v, rtol=1e-4)
+
+    def func2(x):
```

Comment:
Test more functions:

something with orthogonalize tt cores (with rank larger than nessesary and with a normal one)
||Px y||^2 w.r.t. x and y (again with different ranks)
||round||
norm of pseudoinverse of a low rank matrix